### ISSUE TITLE
Run cargo-semver-checks in parallel on all packages

### ISSUE THREAD

**Issue by MarcoIeni** [MarcoIeni/release-plz#645]
## Motivations

cargo-semver-checks requires some time to run.
We could speed things up by running it in parallel on all packages.

## Solution

semver checks is run [here](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L444).

## Alternatives

One could thing that we could run all the package update process in parallel. However at the moment we can't because we run git to checkout at different commits. So running it at the same time in the same project conflicts.

**Comment by IbrahimBagalwa** [comment]
Hello @MarcoIeni 
 I'm interested in contributing to this open source project. However, I'm not entirely clear on what this issue entails and would greatly appreciate some clarification.

Could you please provide me with more information about the issue and what specifically needs to be done to resolve it? I'm eager to learn and contribute in any way I can.

**Comment by MarcoIeni** [comment]
> I'm interested in contributing to this open source project.

Hey Ibrahim 👋  great to hear that! I'll do my best to make it happen. 💪 

I will try to clarify.

[Here](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L288) release-plz calculates the difference between the local package and the one published in the cargo registry.
Inside the `get_diff` function we [run](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L444) cargo-semver-checks, that checks if the public API of the package changed.

As you can see from [this](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L287) for loop, we run the get_diff function on every package (not in parallel). 

What I want to achieve with this issue is to extract the cargo-semver-check run from the `get_diff` function and run this check in parallel on the `packages_to_update` [after](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L307) the for loop. Do you want to give this a try?

As a first step, you could make a PR where you run the cargo-semver-check on the `packages_to_update` in a for loop (not in parallel). Then, once we merge this PR you can raise another PR where you run the check in parallel on all the packages.

What do you think? Let me know if I need to clarify more :)

**Comment by MarcoIeni** [comment]
Wait, actually it's not possible to run this diff after the for loop. Let me think 🤔

**Comment by MarcoIeni** [comment]
The first PR could be to extract cargo-semver-checks from the `get_diff` function and run it [here](https://github.com/MarcoIeni/release-plz/blob/8a87d56c704f41c18353ce94c3c8e2e009ef2e62/crates/release_plz_core/src/next_ver.rs#L289) only if `diff.should_update_version()` is true.
In this way, we run cargo-semver-checks only on the changed packages (not in every package).
After that, we have to refactor the for loop, so that it is possible to run cargo-semver-checks in parallel.

What do you think? Let me know if I need to clarify more or if you need some help :)

**Comment by dutroctu** [comment]
@MarcoIeni I got your idea. May I create the change now

**Comment by MarcoIeni** [comment]
Now we run semver check only in the packages thst changed.
Can we run it in parallel?
